docs/src/pages/components/data-grid/group-pivot/group-pivot.md

```diff
@@ -6,6 +6,191 @@ title: Data Grid - Group & Pivot
+
+#### For all columns
+
+To fully disable the grouping feature, set the `disableRowGrouping` prop to `true`.
+
+It will disable all the features related to the row grouping, even if a model is provided.
+
+{{"demo": "pages/components/data-grid/group-pivot/RowGroupingDisabled.js", "bg": "inline", "defaultCodeOpen": false}}
+
+#### For some columns
+
+To block the grouping of certain columns, set the `groupable` property of `GridColDef` to `false`.
+In the example below, the `director` column can not be grouped. And in all example, the `title` and `gross` columns can not be grouped.
+
+{{"demo": "pages/components/data-grid/group-pivot/RowGroupingColDefCanBeGrouped.js", "bg": "inline", "defaultCodeOpen": false}}
+
+### Grouping columns
+
+#### Single grouping column
+
+By default, the grid will create only one grouping column even if you have several grouping criteria.
+
+{{"demo": "pages/components/data-grid/group-pivot/RowGroupingSingleGroupingCol.js", "bg": "inline", "defaultCodeOpen": false}}
+
+#### Multiple grouping column
+
+To have a grouping column for each grouping criteria, set the `rowGroupingColumnMode` prop to `multiple`.
+
+{{"demo": "pages/components/data-grid/group-pivot/RowGroupingMultipleGroupingCol.js", "bg": "inline", "defaultCodeOpen": false}}
+
+#### Custom grouping column
+
+Use the `groupingColDef` prop to customize the rendering of the grouping column. You can override any property of the `GridColDef` interface except the `field`, the `type` and the properties related to the edition.
+
+If you want to apply your overrides to every grouping column, use the object format of `groupingColDef`.
+
+{{"demo": "pages/components/data-grid/group-pivot/RowGroupingCustomGroupingColDefObject.js", "bg": "inline", "defaultCodeOpen": false}}
+
+If you want to only override properties of certain grouping columns or to apply different overrides based on the current grouping criteria, use the callback format of `groupingColDef`.
+It will be called for each grouping column with the fields of the columns used to build it.
+
+{{"demo": "pages/components/data-grid/group-pivot/RowGroupingCustomGroupingColDefCallback.js", "bg": "inline", "defaultCodeOpen": false}}
+
+#### Show values for the leaves
+
+By default, the leaves nodes don't render anything for their grouping cell.
+
+If you want to display some value, you can provide a `leafField` property to the `groupingColDef`.
+
+{{"demo": "pages/components/data-grid/group-pivot/RowGroupingLeafWithValue.js", "bg": "inline", "defaultCodeOpen": false}}
+
+#### Hide the descendant count
+
+You can use the `hideDescendantCount` property of the `groupingColDef` to hide the amount of descendant of a grouping row.
+
+{{"demo": "pages/components/data-grid/group-pivot/RowGroupingHideDescendantCount.js", "bg": "inline", "defaultCodeOpen": false}}
+
+### Complex grouping value
+
+In most scenarios, when you need to handle complex values, you provide a `valueGetter` property to your column definition.
```

Comment:
The `keyGetter` is a new property on the column definition object.
It's analogous to the known `valueGetter`, which is used to handle objects and other complex values.

The difference between the two is that the `keyGetter` is designed to provide an specific value for grouping, without requiring you to change the displayed value (`valueGetter`).

Reply:
I rephrased with @m4theushw proposal first, which has the same content but with a better formulation.
I think we need to decide for this `valueGetter` => `keyGetter` chaining or not before writing the final sentence.

Reply:
What do you mean by "chaining"?

A different topic: isn't `keyGetter` a too broad name? Putting `valueGetter` aside and there's no clue of the difference between them. Should it have "group" in the name?

Reply:
We had a discussion with @joserodolfofreitas about whether the value passed to `keyGetter` should be the value returned by `valueGetter` or not.

We can rename it `groupingKeyGetter` if it seems clearer.

Reply:
> Suppose we have a "City" column composed of city + state (e.g. "Los Angeles, CA"), then it could be grouped either by the city name or the state.

It adds an additional layer of complexity.
Nothing impossible to manage, we could add a `groupingCriteria` property on `GridColDef`.
But I would wait for users needs for that one.

Reply:
I agree the keyGetter as a name could be improved with "groupingKeyGetter", or even better with "groupingValueGetter", so the correlation with the valueGetter usage is more clear.

Reply:
In which case I guess we are removing the chaining and instead passing the raw value to `groupingValueGetter` right ?

Reply:
Yeah, I think that would make the most sense, the chaining doesn't feel like an intuitive assumption to make, in the sense that users who didn't read a specific line on the docs might get caught off guard.